pkg/reconciler/taskrun/taskrun.go

```diff
@@ -159,6 +160,28 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 		return merr.ErrorOrNil()
 	}
+
+	// If the TaskRun is cancelled, kill resources and update status
+	if tr.IsCancelled() {
+		before := tr.Status.GetCondition(apis.ConditionSucceeded)
+		message := fmt.Sprintf("TaskRun %q was cancelled", tr.Name)
+		err := c.failTaskRun(tr, "TaskRunCancelled", message)
```

Comment:
nit: suggest using the constant `TaskRunSpecStatusCancelled` here instead of literal string `"TaskRunCancelled"`.  Or putting it into a named constant if you want to keep `task.spec.status` conceptually separate from `task.status`.

Reply:
Good idea, I'll fix that, thank you